packages/amqp/lib/AbstractAmqpPublisher.ts

```diff
@@ -42,6 +44,27 @@ export abstract class AbstractAmqpPublisher<MessagePayloadType extends object>
     }
     resolveSchemaResult.result.parse(message)
 
+    // If it's not initted yet, do the lazy init
+    if (!this.isInitted) {
+      // avoid multiple concurrent inits
+      if (!this.initPromise) {
+        this.initPromise = this.init()
+      }
+
+      /**
+       * it is intentional that this promise is not awaited, that's how we keep the method invocation synchronous
+       * RabbitMQ publish by itself doesn't guarantee that your message is delivered successfully, so this kind of fire-and-forget is not strongly different from how amqp-lib behaves in the first place.
+       */
+      this.initPromise
+        .then(() => {
+          this.publish(message)
+        })
+        .catch((err) => {
+          this.handleError(err)
+        })
+      return
```

Comment:
🟢 is `return` needed?

Reply:
yes, otherwise we continue the synchronous execution and fail with an error, as there is no connection

Reply:
Ah yeah, sorry I misread it 😓